hermes-common/src/main/java/pl/allegro/tech/hermes/domain/filtering/avro/AvroPathPredicate.java

```diff
@@ -70,29 +73,37 @@ public boolean test(final FilterableMessage message) {
 
     private List<Object> select(Object record, ListIterator<String> iter) {
         Object current = record;
-        while (iter.hasNext() && current instanceof GenericRecord) {
-            GenericRecord currentRecord = (GenericRecord) current;
-            String selector = iter.next();
-            Matcher arrayMatcher = ARRAY_PATTERN.matcher(selector);
+        while (iter.hasNext()) {
+            if (current instanceof GenericRecord) {
+                GenericRecord currentRecord = (GenericRecord) current;
+                String selector = iter.next();
+                Matcher arrayMatcher = ARRAY_PATTERN.matcher(selector);
+
+                if (arrayMatcher.matches()) {
+                    String idx = arrayMatcher.group(GROUP_IDX);
+                    selector = arrayMatcher.group(GROUP_SELECTOR);
+
+                    current = currentRecord.get(selector);
+                    if (! (current instanceof GenericArray)) {
+                        return emptyList();
+                    }
+
+                    GenericArray<Object> currentArray = (GenericArray) current;
+                    if (idx.equals(WILDCARD_IDX)) {
+                        return selectMultipleArrayItems(iter, currentArray);
+                    } else {
+                        current = selectSingleArrayItem(Integer.valueOf(idx), currentArray);
+                    }
 
-            if (arrayMatcher.matches()) {
-                String idx = arrayMatcher.group(GROUP_IDX);
-                selector = arrayMatcher.group(GROUP_SELECTOR);
-
-                current = currentRecord.get(selector);
-                if (! (current instanceof GenericArray)) {
-                    return emptyList();
-                }
-
-                GenericArray<Object> currentArray = (GenericArray) current;
-                if (idx.equals(WILDCARD_IDX)) {
-                    return selectMultipleArrayItems(iter, currentArray);
                 } else {
-                    current = selectSingleArrayItem(Integer.valueOf(idx), currentArray);
+                    current = currentRecord.get(selector);
                 }
-
-            } else {
+            } else if (current instanceof HashMap) {
+                Map<Utf8, Object> currentRecord = (HashMap<Utf8, Object>) current;
```

Comment:
Skąd pewność, że zawsze tutaj będzie Utf8 oraz Object ?

Reply:
`Object` zawsze będzie, bo wszystko dziedziczy po `Object`, a kluczem w `Map` zakładam, że będzie `Utf8`, bo tak jest deserializowany string w przypadku JSON'a. W naszym przypadku robimy z JSONa AVRO, więc nie zrobisz bardziej złożonego klucza głównego niż string. A jak robisz liczbę to zrób po prostu array.

Reply:
@lukasz20p I think that's because of [Avro standard](https://avro.apache.org/docs/current/spec.html#Maps):
```
Maps
Maps use the type name "map" and support one attribute:

values: the schema of the map's values.
Map keys are assumed to be strings.

For example, a map from string to long is declared with:

{
  "type": "map",
  "values" : "long",
  "default": {}
}
```

For example, metadata is processed as `Map<Utf8, Utf8> metadata`